STMicroelectronics/STM32L4_L4+/app/az_ulib_dm/src/az_ulib_dm.c

```diff
@@ -134,10 +200,48 @@ static az_result install_in_memory(void* base_address, az_span package_name)
   return AZ_ULIB_TRY_RESULT;
 }
 
+static void* find_available_flash(uint32_t size)
+{
+  void* first_that_fit = (void*)&__dcf_pgk_start;
+  for (int i = 0; i < AZ_ULIB_CONFIG_MAX_DM_PACKAGES; i++)
+  {
+    if (_az_dm_cb->_internal.package_list[i].address != NULL)
+    {
+      uint32_t installed_size
+          = *((uint32_t*)(_az_dm_cb->_internal.package_list[i].address)
+              + _AZ_ULIB_DM_PACKAGE_PREAMBLE_CODE_SIZE);
+      installed_size = (installed_size & 0xFFFFF800) + 0x00000800;
```

Comment:
Why do you round it to the nearest 2k?

Reply:
Because we erase the memory before write the new package, if we do not round it, we may end up deleting the end of the previous package.